pyproject.toml

```
@@ -193,21 +179,22 @@ envlist = sqlalchemy{ 1.3, 1.4, 2.0, 2.0-sqlmodel }-{ asyncpg, noasyncpg }-{aws_
 
 [testenv]
 passenv = CI
-deps =
-    sqlalchemy1.3: sqlalchemy<1.4
-    sqlalchemy1.4: sqlalchemy>=1.4,<2
-    sqlalchemy2.0: sqlalchemy>=2
-    sqlalchemy2.0-sqlmodel: sqlalchemy>=2
-    sqlalchemy2.0-sqlmodel: sqlmodel
-    asyncpg: asyncpg
-    aws_rds_iam: boto3
-    pydantic1: pydantic<2
-    pydantic2: pydantic>=2
-
-extras =
-    tests
-
-commands = pytest -vv --cov={envsitepackagesdir}/fastapi_sqla --cov-report xml --cov-report html --junitxml=test-reports/pytest/junit.xml
+skip_install = true
+allowlist_externals = poetry
+commands_pre =
+    poetry install
+    sqlmodel: poetry install --extras "sqlmodel"
+    aws_rds_iam: poetry install --extras "aws_rds_iam"
+    asyncpg: poetry install --extras "asyncpg"
+    asyncpg-aws_rds_iam: poetry install --extras "asyncpg aws_rds_iam"
+    sqlmodel-noasyncpg-aws_rds_iam: poetry install --extras "sqlmodel aws_rds_iam"
+    sqlmodel-asyncpg-noaws_rds_iam: poetry install --extras "sqlmodel asyncpg"
```

a bit similar to [this comment](https://github.com/dialoguemd/fastapi-sqla/pull/138#discussion_r1644880497), should we do
```suggestion
    sqlmodel-aws_rds_iam: poetry install --extras "sqlmodel aws_rds_iam"
    sqlmodel-asyncpg: poetry install --extras "sqlmodel asyncpg"
```

---

I'll try but I don't think this will work because of my mental model of how the badly documented `commands_pre` works.

---

It does work 😅.

---

https://github.com/dialoguemd/fastapi-sqla/pull/138/commits/0679b79d1a4e497de1755218fafd9a8fdf8de8e8